crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -409,85 +403,136 @@ impl Format<PyFormatContext<'_>> for FormatTrailingEndOfLineComment<'_> {
 ///   unnecessary allocations.
 /// * If the content is modified then make as few allocations as possible and use
 ///   a dynamic text element at the original slice's start position.
-pub(crate) const fn format_normalized_comment(
-    comment: Cow<'_, str>,
-    range: TextRange,
-) -> FormatNormalizedComment<'_> {
-    FormatNormalizedComment { comment, range }
+const fn format_normalized_comment(comment: NormalizedComment) -> FormatNormalizedComment {
+    FormatNormalizedComment { comment }
 }
 
-pub(crate) struct FormatNormalizedComment<'a> {
-    comment: Cow<'a, str>,
-    range: TextRange,
+struct FormatNormalizedComment {
+    comment: NormalizedComment,
 }
 
-impl Format<PyFormatContext<'_>> for FormatNormalizedComment<'_> {
+impl Format<PyFormatContext<'_>> for FormatNormalizedComment {
     fn fmt(&self, f: &mut Formatter<PyFormatContext>) -> FormatResult<()> {
         match self.comment {
-            Cow::Borrowed(borrowed) => {
-                source_text_slice(TextRange::at(self.range.start(), borrowed.text_len())).fmt(f)
+            NormalizedComment::Verbatim(text) => source_text_slice(text).fmt(f),
+            NormalizedComment::Body(text) => {
+                write!(f, [token("#"), space(), source_text_slice(text)])
             }
+        }
+    }
+}
 
-            Cow::Owned(ref owned) => {
-                write!(
-                    f,
-                    [
-                        text(owned, Some(self.range.start())),
-                        source_position(self.range.end())
-                    ]
-                )
+#[derive(Debug)]
+enum NormalizedComment {
+    /// A complete comment, including the leading `#`.
+    Verbatim(TextRange),
+
+    /// A comment body, exclusive of the leading `#` and a leading space.
+    Body(TextRange),
+}
+
+impl NormalizedComment {
+    /// Return the width of the comment.
+    fn width(&self, source: &str, indent_width: IndentWidth) -> u32 {
```

Comment:
It'd be nice if this returned `Width`, but there's no way to add `2u32` to a `Width`. (`Width` doesn't even expose a public constructor, which I assumed was intentional.)

Reply:
Could you impl Add for TextWidth and add `TextWidth::from_text("# ")` instead?

Reply:
We may want to consider that `TextWidth::from_text` probably leads to significantly more complicated code because I doubt that the compiler can see past the `unicode_width` crate. I think `2` is fine with a comment